src/action/common/configure_init_service.rs

```diff
@@ -37,6 +41,44 @@ impl ConfigureInitService {
         init: InitSystem,
         start_daemon: bool,
     ) -> Result<StatefulAction<Self>, ActionError> {
+        match init {
+            #[cfg(target_os = "macos")]
+            InitSystem::Launchd => {
+                // No plan checks, yet
+            },
+            #[cfg(target_os = "linux")]
+            InitSystem::Systemd => {
+                // TODO: once we have a way to communicate interaction between the library and the
+                // cli, interactively ask for permission to remove the file
+
+                // NOTE: Check if the service file already exists...
+                if Path::new(SERVICE_DEST).exists() {
+                    return Err(ActionError::FileExists(PathBuf::from(SERVICE_DEST)));
+                }
+                // NOTE: ...and if there are any overrides in the most well-known places for systemd
+                if Path::new(&format!("{SERVICE_DEST}.d")).exists() {
+                    return Err(ActionError::DirExists(PathBuf::from(format!(
+                        "{SERVICE_DEST}.d"
+                    ))));
+                }
+
+                // NOTE: Check if the socket file already exists...
+                if Path::new(SOCKET_DEST).exists() {
+                    return Err(ActionError::FileExists(PathBuf::from(SOCKET_DEST)));
```

Comment:
I think if it already exists, we can check the contents are what we would write, and if so, mark this completed and move on?